### ISSUE TITLE
infinite loop in linkify when email address follows anchor tag.

### ISSUE THREAD

**Issue by thraxil** [mozilla/bleach#154]
Running this:

```
import bleach
text = u'<a href=http://example.com/></a>e@example.com'
print(bleach.linkify(text, parse_email=True))
```

with bleach 1.4.1, html5lib 0.99999, on python 2.7.6 or python 3.4 (tested on Ubuntu 14.04) goes into an infinite loop. 100% CPU and slowly climbing memory.

The email address doesn't have to be immediately following the anchor tag. There can be other text in between.

Turning `parse_email` off avoids the issue.

**Comment by thraxil** [comment]
It looks like this is a duplicate of #124 and is fixed in the current version in git.

Can I *strongly* suggest pushing a release with this fix out to PyPI? In the current state, one can basically DOS a server just by writing an email address in a comment.

**Comment by vivazzi** [comment]
I have this infinite loop too. Bleach version - 1.4.1.
my example:
```>>> bleach.linkify('<p>test<br/>, email test@mail.ru</p>')```

In debug I have seen next result like this:
```u'" ><a href="<a href="mailto:test@mail.ru" >mailto:test@mail.ru</a>" >test@mail.ru</a>'```
It likes this recursion.

Alhought, next test is passed:
```
>>> bleach.linkify('<p>test email test@mail.ru</p>')
u'<p>test email <a href="mailto:test@mail.ru">test@mail.ru</a></p>'
```

**Comment by ebrelsford** [comment]
I am running into the same issue with 1.4.1 but not with master. Please do create a new release!

**Comment by msom** [comment]
+1 for a new release

**Comment by willkg** [comment]
I can't reproduce this issue with master tip and it talks about bleach 1.4.2 having a fix for is, so I'm going to assume that this was fixed in 1.4.2 and close it out.

If you can still reproduce it, please let me know.
